src/commands/newFile.ts

```diff
@@ -0,0 +1,74 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the VS Code Swift open source project
+//
+// Copyright (c) 2021-2024 the VS Code Swift project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of VS Code Swift project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import * as fs from "fs/promises";
+import * as path from "path";
+import * as vscode from "vscode";
+import { fileExists, pathExists } from "../utilities/filesystem";
+
+const extension = "swift";
+const defaultFileName = `Untitled.${extension}`;
+
+export async function newSwiftFile(
+    uri?: vscode.Uri,
+    isDirectory: (uri: vscode.Uri) => Promise<boolean> = async uri => {
+        return (await vscode.workspace.fs.stat(uri)).type === vscode.FileType.Directory;
+    }
+) {
+    if (uri) {
+        // Attempt to create the file at the given directory.
+        const dir = (await isDirectory(uri)) ? uri.fsPath : path.dirname(uri.fsPath);
+        const defaultName = path.join(dir, defaultFileName);
+        const givenPath = await vscode.window.showInputBox({
```

Comment:
https://code.visualstudio.com/api/references/vscode-api#window showOpenDialog or showSaveDialog is preferred way. User can enable simple file dialog in settings if they want to type out

Reply:
Much better, thanks